language/stdlib/modules/block.mvir

```diff
@@ -0,0 +1,48 @@
+module Block {
+
+  resource T {
+    // Height of the current block
+    height: u64,
+  }
+
+  // This can only be invoked by the Association address, and only a single time.
+  // Currently, it is invoked in the genesis transaction
+  public initialize() {
+    // Only callable by the Association address
+    assert(get_txn_sender() == 0xA550C18, 1);
+
+    move_to_sender<T>(T { height: 0 });
+    return;
+  }
+
+  // Set the metadata for the current block.
+  // The runtime always runs this before executing the transactions in a block.
```

Comment:
Who signs for this transaction? No one? 

Will it be another part of the VM? (added to the current responsibility of calling account prologue/epilogue?

Reply:
Correct. @zekun000's design doc has some more details on this (will send to you).